apollo-compiler/src/test/graphql/com/example/enums_with_reserved_words/TestOperation.graphql

```diff
@@ -0,0 +1,7 @@
+# The generated enum class should contain all members
+query TestQuery {
+  enum
+  ... on Query {
+    enum
+  }
```

Comment:
Nit: the inline fragment is not needed, is it?

Reply:
Sorry I meant to explain and then forgot 😅.  This is to trigger the operationBased codegen in `CodegenTest` (it's only triggered when there's fragments).

Reply:
Since this only tests enums codegen, I'd say we don't need to generate operationBased code? Saves CI time and a few test fixtures?

In general, I think we reach the point where we can't really test the matrix of all input parameters because they are too many:

* codegenModels
* targetLanguage
* sealedClassesAsEnumMatching
* languageVersion
* etc...

If we were to have 100% code coverage, compilation time would grow exponentially so we're bound to make focused tests for paramters like this

Reply:
Oh yeah totally agree - wasn't 100% sure the enum generation was the same for both codegens but since it is, this is unneeded 👍